diff --git a/sql/anon.sql b/sql/anon.sql
index 0cdc769..c004139 100644
--- a/sql/anon.sql
+++ b/sql/anon.sql
@@ -1141,3 +1141,9 @@ $$
 -- TODO : https://en.wikipedia.org/wiki/L-diversity
 
 -- TODO : https://en.wikipedia.org/wiki/T-closeness
+
+-- NEON Patches
+
+GRANT ALL ON SCHEMA anon to neon_superuser;
+GRANT ALL ON ALL TABLES IN SCHEMA anon TO neon_superuser;
+
diff --git a/sql/init.sql b/sql/init.sql
index 7da6553..0c5f6c9 100644
--- a/sql/init.sql
+++ b/sql/init.sql
@@ -74,6 +74,26 @@ $$
 
 SECURITY LABEL FOR anon ON FUNCTION anon.load_csv IS 'UNTRUSTED';
 
+CREATE OR REPLACE FUNCTION anon.load_csv_superuser(
+  dest_table REGCLASS,
+  csv_file TEXT
+)
+RETURNS BOOLEAN AS
+$$
+BEGIN
+  RETURN anon.load_csv(dest_table, csv_file);
+END;
+$$
+  LANGUAGE plpgsql
+  VOLATILE
+  RETURNS NULL ON NULL INPUT
+  PARALLEL UNSAFE -- because of the EXCEPTION
+  SECURITY DEFINER
+  SET search_path=''
+;
+
+SECURITY LABEL FOR anon ON FUNCTION anon.load_csv_superuser IS 'UNTRUSTED';
+
 -- load fake data from a given path
 CREATE OR REPLACE FUNCTION anon.init(
   datapath TEXT
