diff --git a/regress/core/tests.mk.in b/regress/core/tests.mk.in
index 3dd65b9..2735398 100644
--- a/regress/core/tests.mk.in
+++ b/regress/core/tests.mk.in
@@ -143,8 +143,7 @@ TESTS += \
 	$(top_srcdir)/regress/core/oriented_envelope \
 	$(top_srcdir)/regress/core/point_coordinates \
 	$(top_srcdir)/regress/core/out_geojson \
-  $(top_srcdir)/regress/core/wrapx \
-	$(top_srcdir)/regress/core/computed_columns
+        $(top_srcdir)/regress/core/wrapx
 
 # Slow slow tests
 TESTS_SLOW = \
